common/containers/TabSection/Notifications.tsx

```diff
@@ -5,28 +5,38 @@ import {
 } from 'actions/notifications';
 import React from 'react';
 import { connect } from 'react-redux';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
 import NotificationRow from './NotificationRow';
 import './Notifications.scss';
 
 interface Props {
   notifications: Notification[];
   closeNotification: TCloseNotification;
 }
+
+const Transition = props => (
+  <CSSTransition
+    {...props}
+    classNames="animation"
+    timeout={{ enter: 500, exit: 500 }}
+  />
+);
+
 export class Notifications extends React.Component<Props, {}> {
   public render() {
-    if (!this.props.notifications.length) {
-      return null;
-    }
     return (
-      <div className="Notifications">
-        {this.props.notifications.map((n, i) => (
-          <NotificationRow
-            key={`${n.level}-${i}`}
-            notification={n}
-            onClose={this.props.closeNotification}
-          />
-        ))}
-      </div>
+      <TransitionGroup className="Notifications">
+        {this.props.notifications.map(n => {
+          return (
+            <Transition key={n.id}>
+              <NotificationRow
+                notification={n}
+                onClose={this.props.closeNotification}
+              />
+            </Transition>
+          );
+        })}
+      </TransitionGroup>
```

Comment:
I haven't looked at the latest `react-transition-group` in a while, but I'm pretty sure having both of these transition components is overkill. I think all you want is `CSSTransition`. Their [example of CSSTransition](https://codesandbox.io/s/yv645oq21x) doesn't use `TransitionGroup` at all.

Reply:
[TransitionGroup](https://reactcommunity.org/react-transition-group/#TransitionGroup) is used to manage multiple `CSSTransition` or `Transition` elements. It's useful when animating items in a list, like a list of notifications. For animating individual elements, like in their [example of CSSTransition](https://codesandbox.io/s/yv645oq21x), `TransitionGroup` isn't needed, but when you want to animate a list of transitions it is.